### PR TITLE
escape windows in bun upgrade

### PR DESCRIPTION
**Pull request by paperclover**
### What does this PR do?

Fixes https://github.com/oven-sh/bun/issues/12872
Fixes https://github.com/oven-sh/bun/issues/12898

I added an escaping test to verify the escaping rules are reasonable.

### CODE REVIEW

**Comment by robobun** [comment]
@paperdave, your commit 7df608e633ade26d86ef83de592d079b953f311d is building: [`#1059`](https://buildkite.com/bun/bun/builds/1059)

File: test/cli/install/bun-upgrade.test.ts [hunk 1]
```diff
@@ -176,3 +176,13 @@ it("zero arguments, should succeed", async () => {
   // Should not contain error message
   expect(stderr.toString()).not.toContain("error:");
 });
+
+import { escapePowershell } from "bun:internal-for-testing";
```
Comment (Electroid) [inline test/cli/install/bun-upgrade.test.ts]: Can we maybe move this to a new `test/unit` folder? Having these orthogonal tests glittered around behaviour tests isn't great.

Comment (paperclover) [reply test/cli/install/bun-upgrade.test.ts]: sure

File: src/cli/upgrade_command.zig [hunk 2]
```diff
@@ -652,10 +652,10 @@ pub const UpgradeCommand = struct {
                     // Run a powershell script to unzip the file
                     const unzip_script = try std.fmt.allocPrint(
                         ctx.allocator,
-                        "$global:ProgressPreference='SilentlyContinue';Expand-Archive -Path {s} {s} -Force",
+                        "$global:ProgressPreference='SilentlyContinue';Expand-Archive -Path \"{}\" \"{}\" -Force",
```
Comment (Jarred-Sumner) [inline src/cli/upgrade_command.zig:655]: can these quotes be done in the formatter? I think it's extra confusing this way

Comment (paperclover) [reply src/cli/upgrade_command.zig:655]: ok

**Pull request merged**